src/ui/popup.js

```diff
@@ -272,7 +272,12 @@ class Popup extends Evented {
             }
         }
 
-        const offsetedPos = this._pos.add(offset[anchor]).round();
+        let offsetedPos = this._pos.add(offset[anchor]);
+        //snap to interger offset if this is last tick of an easing
+        //animation, or if this event didn't come from easing loop
+        if (ev && (ev.finalEasingTick === true || ev.easing !== true)) {
```

Comment:
[`Marker` has similar logic](https://github.com/mapbox/mapbox-gl-js/blob/213d18a501fd505a846448599ca57d56a9469967/src/ui/marker.js#L153-L153), but the condition is based on checking the event type (`move` vs `moveend`). Will that work here too, instead of adding new properties to the event object?